src/details/ArborX_Polytope.hpp

```diff
@@ -0,0 +1,270 @@
+/****************************************************************************
+ * Copyright (c) 2012-2020 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+#ifndef ARBORX_POLYTOPE_HPP
+#define ARBORX_POLYTOPE_HPP
+
+#include <ArborX_DetailsKokkosExt.hpp> // ArithmeticTraits
+#include <ArborX_DetailsTags.hpp>
+#include <ArborX_Point.hpp>
+
+#include <Kokkos_Macros.hpp>
+
+namespace ArborX
+{
+
+namespace Details
+{
+struct Direction
+{
+  float _data[3];
+};
+KOKKOS_FUNCTION float project(Point const &p, Direction const &d)
+{
+  float r = 0.;
+  for (int i = 0; i < 3; ++i)
+  {
+    r += p[i] * d._data[i];
+  }
+  return r;
+}
+} // namespace Details
+
+template <typename T>
+struct Polytope : public T
+{
+  static constexpr int n_directions = T::directions.size();
+  Kokkos::Array<float, n_directions> _min_values;
+  Kokkos::Array<float, n_directions> _max_values;
```

Comment:
I would make them pairs, as the typical thing is to do `if (v < min || v > max)`.